src/modules/main/navbar/user-section/user-details/UserDetails.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { Avatar } from '@material-ui/core';
+
+import { Button, Img } from 'ui';
+
+import { TemplateCategory } from 'core/api';
+import { useAuthProvider } from 'core/auth';
+
+import csx from './UserDetails.scss';
+
+const MAX_VISIBLE_TECHNOLOGIES = 16;
+
+const UserDetails = () => {
+  const {
+    user: {
+      username,
+      email,
+      company,
+      seniority,
+      technologies,
+      yearsOfExperience,
+      position,
+      connectedWithGithub
+    },
+    logOut
+  } = useAuthProvider();
+
+  const { replace } = useHistory();
+
+  const DETAILS: Record<string, string | number> = {
```

Comment:
You should create function outside component - and memoize result.
fn should take user object and return all needed labels, object keys, and values - much easier to read and also performance benefit